pkg/nimble/netif/nimble_netif.c

```diff
@@ -331,6 +336,27 @@ static void _on_data(nimble_netif_conn_t *conn, struct ble_l2cap_event *event)
     ble_l2cap_recv_ready(event->receive.chan, rxb);
 }
 
+#if IS_USED(MODULE_GNRC_NETAPI_NOTIFY)
+/**
+ * @brief   Sends a netapi notification for a connection event.
+ *
+ * @param[in] notify    The type of notification event.
+ * @param[in] addr      BLE address of the node that (dis-)connected.
+ */
+static inline void _dispatch_connection_event(netapi_notify_t notify, uint8_t *addr)
+{
+    netapi_notify_l2_connection_t event = {
+        .l2addr_len = BLE_ADDR_LEN,
+        .if_pid = _netif.pid,
+    };
+
+    memcpy(event.l2addr, addr, BLE_ADDR_LEN);
+
+    gnrc_netapi_notify(GNRC_NETTYPE_L2_DISCOVERY, GNRC_NETREG_DEMUX_CTX_ALL,
+                       notify, &event, sizeof(netapi_notify_l2_connection_t));
+}
+#endif /* MODULE_GNRC_NETAPI_NOTIFY */
```

Comment:
How about

```suggestion
#else 
static inline void _dispatch_connection_event(netapi_notify_t notify, uint8_t *addr)
{
    (void)notify;
    (void)addr;
}
#endif /* MODULE_GNRC_NETAPI_NOTIFY */
```

to not uglify the code with all those `#if` blocks